src/main/java/com/sarahhand/fractals/mapper/ColorPaletteMapper.java

```
@@ -0,0 +1,109 @@
+package com.sarahhand.fractals.mapper;
+
+import java.awt.Color;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import com.sarahhand.fractals.model.ColorPalette;
+
+public class ColorPaletteMapper{
+	
+	private static final ColorPaletteMapper defaultMapper = new ColorPaletteMapper();
+	
+	public static ColorPaletteMapper getDefaultMapper(){
+		return defaultMapper;
+	}
+	
+	private ColorPaletteMapper(){}
+	
+	public ColorPalette map(List<Color> colors, List<Integer> positions){
+		
+		if(errorCases(colors, positions)){
+			return null;
+		}
+		
+		Map<Integer, Color> map = new HashMap<>();
+		
+		Color[] palette = new Color[ColorPalette.COLOR_PALETTE_LENGTH];
+		
+		for(int n = 0; n < colors.size(); n++){
+			map.put(positions.get(n), colors.get(n));
+		}
+		
+		int beginInterval = 0;
+		
+		while(beginInterval < ColorPalette.COLOR_PALETTE_LENGTH){
+
+			int endInterval = beginInterval + 1;
+			
+			while(!positions.contains(endInterval)){
+				endInterval++;
+			}
+			
+			Color[] gradient = gradiate(map.get(beginInterval), map.get(endInterval), endInterval-beginInterval);
+			
+			System.arraycopy(gradient, 0, palette, beginInterval, gradient.length);
+			
+			beginInterval = endInterval;
+		}
+		
+		return new ColorPalette(palette);
+	}
+	
+	private boolean errorCases(List<Color> colors, List<Integer> positions){
```

I think this method should throw exceptions for each check that fails instead of just returning true/false.  The reason I think this is because otherwise no meaningful information is provided as to why the mapping failed.